pkg/utils/yaml.go

```diff
@@ -1,142 +1,101 @@
-package install
+package utils
 
 import (
-	"bytes"
 	"context"
 	"fmt"
-	"io"
 	"path/filepath"
 	"strconv"
 	"strings"
-	"time"
-
-	"golang.org/x/sync/errgroup"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/cli-runtime/pkg/genericclioptions"
-	"k8s.io/client-go/kubernetes"
-	storagev1 "k8s.io/client-go/kubernetes/typed/storage/v1"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/remotecommand"
+
 	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative/pkg/manifest"
 	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
 	"sigs.k8s.io/yaml"
 )
 
-// NewClientConfig returns a client-go rest config
-func NewClientConfig() (*rest.Config, error) {
-	configFlags := &genericclioptions.ConfigFlags{}
-
-	config, err := configFlags.ToRESTConfig()
+// GetManifestFromMultiDoc returns an individual object string from a multi-doc yaml file
+// after searching by kind. Note: the first object in multiManifest matching kind is returned.
+func GetManifestFromMultiDoc(multiDoc, kind string) (string, error) {
+	objs, err := manifest.ParseObjects(context.TODO(), multiDoc)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
-	return config, nil
-}
-
-// GetClientsetFromConfig returns a k8s clientset from a client-go rest config
-func GetClientsetFromConfig(config *rest.Config) (*kubernetes.Clientset, error) {
-	clientset, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return nil, err
+	for _, obj := range objs.Items {
+		if obj.UnstructuredObject().GetKind() == kind {
```

Comment:
`if obj.UnstructuredObject().GetKind() != kind { continue }` should be a bit more goish.

Reply:
Done